e2e/single-cluster/imagescan_test.go

```diff
@@ -92,3 +68,211 @@ var _ = Describe("Image Scan", Label("infra-setup"), func() {
 		})
 	})
 })
+
+var _ = Describe("Image Scan dynamic tests pushing to ttl.sh", Label("infra-setup"), func() {
+	var (
+		clonedir   string
+		k          kubectl.Command
+		gh         *githelper.Git
+		repository *git.Repository
+		assetdir   string
+		tmpRepoDir string
+		image      string
+		imageTag   string
+	)
+
+	JustBeforeEach(func() {
+		k = env.Kubectl.Namespace(env.Namespace)
+		tmpdir := GinkgoT().TempDir()
+		clonedir = path.Join(tmpdir, "clone")
+		repository = setupRepo(k, tmpdir, clonedir, tmpRepoDir)
+	})
+
+	AfterEach(func() {
+		_, _ = k.Delete("gitrepo", "imagescan")
+		_, _ = k.Delete("secret", "git-auth")
+	})
+	When("deploy imagescan setup with pre-release images", func() {
+		BeforeEach(func() {
+			assetdir = "imagescan/pre-releases-ok"
+			tmpRepoDir = GinkgoT().TempDir()
+			image, imageTag = initRegistryWithImageAndTag("k8s.gcr.io/pause", "0.0.0-40")
+			applyTemplateValues(assetdir, tmpRepoDir, image, imageTag)
+		})
+		It("updates the docker reference", func() {
+			By("checking the deployment exists")
+			Eventually(func() string {
+				out, _ := k.Namespace("default").Get("pods")
+				return out
+			}).Should(ContainSubstring("pause-prerelease-"))
+
+			By("checking the bundle has the original image tag")
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, imageTag)))
+
+			By("pushing a new tag to the registry and checking the new image tag is found in the bundle")
+			newTag := "0.0.0-50"
+			imageTag = tagAndPushImage("k8s.gcr.io/pause", image, newTag)
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, imageTag)))
+
+			By("checking that the new tag is pushed in the git repository")
```

Comment:
This is a thorough check, but arguably if the new image tag has made it into the bundle, then that would mean that it had first been propagated to the git repository.